sql/hive/src/main/scala/org/apache/spark/sql/hive/hiveUDFs.scala

```diff
@@ -157,10 +157,15 @@ private[hive] case class HiveGenericUDF(
     val setValues = evals.zipWithIndex.map {
       case (eval, i) =>
         s"""
-           |if (${eval.isNull}) {
-           |  $refEvaluator.setArg($i, null);
-           |} else {
-           |  $refEvaluator.setArg($i, ${eval.value});
+           |try {
+           |  ${eval.code}
+           |  if (${eval.isNull}) {
+           |    $refEvaluator.setArg($i, null);
+           |  } else {
+           |    $refEvaluator.setArg($i, ${eval.value});
+           |  }
```

Comment:
1. It seems that `catch (...)` is only related to `${eval.code}`
2.Why is it `Exception` instead of `Throwable`?

Reply:
>> It seems that catch (...) is only related to ${eval.code}

Yes, we should catch the exceptions from child's executions.

>> Why is it Exception instead of Throwable?

The Exception can catch most of throwable problems. It's ok to be change to Throwable.